src/infra/main/template.yaml

```diff
@@ -151,3 +157,311 @@ Resources:
+    Type: AWS::Serverless::Function
+    # checkov:skip=CKV_AWS_116: DLQ is not appropriate for a Lambda invoked by an API
+    # checkov:skip=CKV_AWS_115: Function is not configured for function-level concurrent execution Limit
+    Properties:
+      FunctionName: !Sub '${AWS::StackName}-AccountDeletionProcessorFunction'
+      Description: "A lambda function that updates an account status indicating that the account has been deleted"
+      CodeUri: ../../../dist/handlers/account-deletion-processor-handler
+      Timeout: 10
+      Handler: account-deletion-processor-handler.handler
+      Role: !GetAtt AccountDeletionProcessorFunctionRole.Arn
+      Environment:
+        Variables:
+          TABLE_NAME:
+            Fn::ImportValue: !Sub "${CoreStackName}-AccountStatusTableName"
+          DELETED_ACCOUNT_STATUS_RETENTION_SECONDS: 63072000
+          METRIC_SERVICE_NAME: !Sub "${AWS::StackName}-AccountDeletionProcessorFunction"
+      Events:
+        SNSEvent:
+          Type: SNS
+          Properties:
+            RedrivePolicy: !Sub "{ \"deadLetterTargetArn\": \"arn:aws:sqs:${AWS::Region}:${AWS::AccountId}:${AWS::StackName}-AccountDeletionProcessorDLQ\" }"
+            SqsSubscription:
+              BatchSize: 1
+              Enabled: true
+              QueueArn: !GetAtt AccountDeletionProcessorQueue.Arn
+              QueueUrl: !Ref AccountDeletionProcessorQueue
+            Topic: !If
+              - IsInternal
+              - !Ref AccountDeletionProcessorSNSTopic
+              - !FindInMap [ EnvConfig, !Ref Environment, AccountDeletionProcessorSNSTopicARN ]
+      Tags:
+        Product: !Ref ProductTagValue
+        System: !Ref SystemTagValue
+        Environment: !Ref Environment
+        Owner: !Ref OwnerTagValue
+        Source: !Ref SourceTagValue
+        CheckovRulesToSkip: CKV_AWS_116.CKV_AWS_115
+
+  AccountDeletionProcessorResourcePolicy:
+    Type: AWS::Lambda::Permission
+    Properties:
+      FunctionName: !Ref AccountDeletionProcessorFunction
+      Action: lambda:InvokeFunction
+      Principal: sns.amazonaws.com
+      SourceAccount: !Ref AWS::AccountId
+      SourceArn: !If
+        - IsInternal
+        - !Ref AccountDeletionProcessorSNSTopic
+        - !FindInMap [ EnvConfig, !Ref Environment, AccountDeletionProcessorSNSTopicARN ]
+
+  AccountDeletionProcessorFunctionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName:  !Sub '${AWS::StackName}-AccountDeletionProcessorFunctionRole'
+      PermissionsBoundary: !If [ UsePermissionsBoundary, !Ref PermissionsBoundary, !Ref AWS::NoValue ]
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action:
+              - "sts:AssumeRole"
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AWSXrayWriteOnlyAccess
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole
+      Policies:
```

Comment:
can we please have all the policy statements under the policy resource?

Reply:
Yeah similar to the vc-storage one, it has to be here for the SQS to connect to the SNS topic in the different environment mappings. This was tested before